test/run_envoy_tests.sh

```diff
@@ -9,10 +9,16 @@ BINARY_DIR=$2
 # https://bazel.build/versions/master/docs/test-encyclopedia.html. In particular, TEST_SRCDIR is
 # where we expect to find the generated outputs of various scripts preparing input data (these are
 # not only the actual source files!).
-: ${TEST_TMPDIR:=/tmp/envoy_test_tmp}
-: ${TEST_SRCDIR:=/tmp/envoy_test_runfiles}
+# It is a precondition that both $TEST_TMPDIR and $TEST_SRCDIR are empty.
+if [ -z "$TEST_TMPDIR" ] || [ -z "$TEST_SRCDIR" ]
+then
+  TEST_BASE=$(mktemp -d /tmp/envoy_test.XXXXXXXX)
```

Comment:
Hmm. Actually, I think this is going to cause problems related to why I need https://github.com/lyft/envoy/pull/627. Basically. we have Lyft private tests that have configuration files that I need to copy into the runfiles directory before this script runs. If TEST_TMPDIR/TEST_SRCDIR is not defined, which it won't be until we finish bazel migration, I'm going to get 2 different directories each time I do the setup.

Not sure the best way of dealing with this? Is it really horrible to just rm -fr in the case where we are using the default directories?